tests/providers/microsoft/azure/test_utils.py

```diff
@@ -77,6 +80,19 @@ def test_func() -> dict[str, Any]:
     assert "workload_identity_tenant_id" in widgets
 
 
+@mock.patch(f"{MODULE}.DefaultAzureCredential")
```

Comment:
@Lee-W, the changes look good. Do you think it is a good idea to elaborate the unit tests a bit by adding mock objects and running tests to compare expected vs. actual for the method calls?

Reply:
I'm not sure. I think the main point here is verifying we're calling the right object

Reply:
Yeah - for me it's enough to check they are called. Too detailed tests are often problematic - especially in case of some auth methods that tend to change when client libraries change. We often have to fix such tests when new versions of libraries are released just to add a new parameter expected etc. So I'd say just "called" is good enough for me - especially that we have detailed test for the credentials class itself

Reply:
Agree, @Lee-W and @potiuk, thanks for clarifying.